### DISCUSSION TITLE
Re-populating filters from elastic search after navigating.

### DISCUSSION THREAD

**Discussion by RukshanJS** [appbaseio/reactivesearch#1922] [Q&A]
I have a screen (say screen A) where Reactive Search is working fine to get some Books let's say. I have 2 filter fields (MULTILIST) with values as follows.

- genre => comic, biography.
- yearPublished => 1950-2000, 2005-2020

What I need to get done,
On another screen (say screen B), I can click a button `Preset1` which has `genre` as `comic`, `yearPublished` as `1950-2000` associated with it. When I click this, I want to navigate to screen A, and apply the values `comic` to `genre` and `1950-2000` to `yearPublished`, and re-render the ReactiveList.

How can I update the redux store/ set the filters from a component outside the scope of ReactiveBase component?

P.S. => I could do this using a component under the ReactiveBase component (using SelectedFilters and its setValue prop), but have problems doing this from a component outside ReactiveBase (because the two screens are in different routes)

**Comment by siddharthlatest** [comment]
@RukshanJS Can you try using the [StateProvider](https://docs.appbase.io/docs/reactivesearch/v3/advanced/stateprovider/) component for doing this? It gives you access to the current state of ReactiveBase, which can then be used in a controlled fashion to set the state of a relevant component.

**Reply by RukshanJS** [reply]
Thanks for the reply! Yes, I'll use it and give you an update.

**Reply by RukshanJS** [reply]
@siddharthlatest I tried using `StateProvider`, but it does not have any method to "SET" the state of the filters programmatically, right? I can "GET" the state but could you please clarify how this can be used to set the state? - I want to be able to firstly have the above mentioned filters applied when I go from screen B to screen A (as in my original post). So when I go from Screen B to Screen A, my filters are applied programmatically and relevant results are already displayed.

**Comment by RukshanJS** [comment]
Ok so I have some progress. I have used the `defaultQuery` prop of `ReactiveList` to display the results from a stored query. Then to update my UI filters (for already selected filters) I'm using a separate state management approach (mentioned in my original post - using `SelectedFilters` component). Still not sure if I'm able to populate the RS filters (UI) from code, but hope my method is alright. If you can share any suggestions, highly appreciated! Thanks. :)
